include/circt/Dialect/RTL/Combinatorial.td

```diff
@@ -163,6 +163,32 @@ def XorROp : UnaryI1ReductionRTLOp<"xorr">;
 // Other Operations
 //===----------------------------------------------------------------------===//
 
+// Extract a range of bits from the specified input.
+def StructExtractOp : RTLOp<"struct_extract", [NoSideEffect]> {
+  let summary = "Extract a named field from a struct.";
+
+  let arguments = (ins StrAttr:$field, AnyRTLStructType:$input);
+  let results = (outs AnyType:$result);
```

Comment:
Please use RTLType instead of AnyType for the result.  My goal/hope is for that to be the unifying predicate that filters the "types that make sense for RTL dialect"

Reply:
Not sure I agree with this given that so far the RTL dialect has adopted an open type system. My expectation would be to allow AnyType then allow passes to lower it into something which "makes sense" for the RTL dialect. If, however, this was the SV dialect I would agree.

Reply:
Huh fair enough.  On reflection, my primary goal is to prevent InOutType from ending up in nested positions, but maybe the cure is worse than the problem.

Reply:
avoiding inout is definately important.  All add something for ops that want to enforce that.

Reply:
FYI `RTLType` prohibits inout.